src/blocks/gallery-offset/edit.js

```diff
@@ -223,26 +224,28 @@ const Edit = ( props ) => {
 							return (
 								<li className={ itemClasses } key={ img.id || img.url }>
 									<GalleryImage
-										url={ img.url }
 										alt={ img.alt }
-										id={ img.id }
-										isSelected={ isSelected && selectedImage === index }
-										onRemove={ onRemoveImage( index ) }
-										onSelect={ onSelectImage( index ) }
-										setAttributes={ ( attrs ) => setImageAttributes( index, attrs ) }
-										caption={ img.caption }
 										aria-label={ ariaLabel }
+										caption={ img.caption }
 										captions={ captions }
-										supportsCaption={ true }
+										gutterCustom={ gutterCustom }
+										id={ img.id }
+										imageIndex={ index }
```

Comment:
This does indeed seem like a reasonable solution given the block's current architecture. I see also how you've edited the GalleryImage component to handle the styles. For a fix like this, we should attempt to fix it from within the GutterWrapper which is responsible for applying _gutter_ related styles in the editor as well as the save function. Here is an example of where custom styles are set specifically within the GutterWrapper component.

https://github.com/godaddy-wordpress/coblocks/blob/fac65d4824fb0c4c64c5a5be0ef5a938ea3a46a1/src/components/gutter-control/gutter-wrapper.js#L39-L41

Another aspect to consider here is that the GalleryImage component is being deprecated and removed from our Gallery blocks in favor of core/image inner blocks following the V2 paradigm. So these changes will soon be undone as we move away from the GalleryImage solution and therefore the fix is better applied in the GutterWrapper.